addons/payment/models/payment_acquirer.py

```diff
@@ -618,6 +629,20 @@ def s2s_get_tx_status(self):
         return True
 
     @api.multi
+    def execute_callback(self):
+        res = None
+        for transaction in self.filtered(lambda tx: tx.callback_ir_model_id and tx.callback_res_id and tx.callback_method):
+            valid_token = transaction._generate_callback_hash()
+            if not consteq(ustr(valid_token), transaction.callback_hash):
+                _logger.warning('Receiving invalid callback for transaction %d' % (transaction.id))
+                continue
+
+            record = self.env[transaction.callback_ir_model_id.model].browse(transaction.callback_res_id).exists()
```

Comment:
@tde-banana-odoo 
Should it fail silently if no record is found? We could probably leave a trace in the log... or raise.